components/Download.vue

```diff
@@ -143,16 +149,69 @@ export default class extends Vue {
           body: body
         }
       );
-      const data = await response.blob();
+      
+      if(response.status === 200) {
+        const data = await response.blob();
 
-      let url = window.URL.createObjectURL(data);
-      window.location.assign(url);
+        let url = window.URL.createObjectURL(data);
+        window.location.assign(url);
+      }
+      else {
+        this.errorMessage = `Status code: ${response.status}, Error: ${response.statusText}`;
+      }
 
       this.isReady = true;
     } catch (err) {
       this.isReady = true;
-      this.errorMessage = err.message || err;
+      this.errorMessage = `Status code: ${err.status}, Error: ${err.statusText}` || err;
+    }
+  }
+
+  public async getGoodIcon(): Promise<any> { 
+    var test;
+    var goodIcon = (this.manifest as any).icons.find(
+      icon => icon.sizes.includes("512") || icon.sizes.includes("192")
+    );
+
+    if(goodIcon) {
+      await this.isValidUrl(goodIcon.src).then(response => {
+        if (response && response.isValidUrl) {
+          return goodIcon;
+        }
+      });
     }
+
+    let i = 0;
+    for (i; i < (this.manifest as any).icons.length; i++) {
+      goodIcon = (this.manifest as any).icons[i];
+      test = await this.isValidUrl(goodIcon.src).then(response => {
+        return response;
+      });
+      if (test && test.isValidUrl) {
+          break;
+      }
+    }
+    if(i === (this.manifest as any).icons.length) {
+      return test;
+    }
+    else {
+      return goodIcon;
+    }
+  }
+  
+  public async isValidUrl(url): Promise<any> { 
+    // CORS Anywhere is a NodeJS proxy which adds CORS headers to the proxied request.
+    const response = await fetch('https://cors-anywhere.herokuapp.com/' + url)
```

Comment:
Does the URL need to be escaped here?

Reply:
I know I would get a comment on this 😉. As you know we are using fetch() function in other places to check if url is valid or not. That technic is not helping here because mode: 'no-cors' is blocking frontend JavaScript code from looking at contents of the response body and headers under all circumstances. So, that is why “cors-anywhere” is used to work around to get response.

Reply:
Ive heard of cors-anywhere before, is there an Azure service we could flip this over to in a future sprint @TesheG ?